api/router.js

```diff
@@ -11,10 +11,14 @@ module.exports = function($window, renderer, pubsub) {
 		var replay = router.defineRoutes(routes, function(payload, args, path, route) {
 			args.path = path, args.route = route
 			if (typeof payload.onmatch === "function") {
+				var resolved = false
```

Comment:
Branched that off to its own thing #1259 since it's a good idea regardless of RouteResolver semantics shenanigans